### PR TITLE
First attempt to implement the connection monitor

### PR DESCRIPTION
**Pull request by pierantoniomerlino**
Signed-off-by: pierantoniomerlino <pierantonio.merlino@eurotech.com>
See #1417

### CODE REVIEW

**Comment by MMaiero** [comment]
@pierantoniomerlino Please be careful that in the meanwhile the paho library changed.

File: kura/org.eclipse.kura.core/OSGI-INF/metatype/org.eclipse.kura.data.DataService.xml [hunk 1]
```diff
@@ -36,6 +36,24 @@
...
             description="Frequency in seconds to retry a connection of the Data Publishers after a disconnect (0 to disable)."/>
             
+        <AD id="connect.retry-attempts"
+            name="connect.retry-attempts"
+            type="Integer"
+            cardinality="0" 
+            required="true"
+            default="60"
+            min="0"
+            description="Number of connection attempts of the Data Publishers before reset."/>
```
Comment (cdealti) [inline kura/org.eclipse.kura.core/OSGI-INF/metatype/org.eclipse.kura.data.DataService.xml]: * before reset -> before give-up
* Set 0 to disable?

**Comment by cdealti** [comment]
@markoer the DataService is a key component that if it fails will cause headaches with customers.
Assuming this will be merged, can you please add some new QA tests for validation covering the corner cases?
